akka-stream/src/main/scala/akka/stream/Materializer.scala

```diff
@@ -154,40 +190,20 @@ object Materializer {
   implicit def matFromSystem(implicit provider: ClassicActorSystemProvider): Materializer =
     SystemMaterializer(provider.classicSystem).materializer
 
-}
+  /**
+   * Scala API: Create a materializer whose lifecycle will be tied to the one of the passed actor. When the actor stops the materializer
+   * will stop and all streams created with it will be failed with an [[AbruptTerminationExeption]]
+   */
+  @silent("deprecated")
```

Comment:
could we reverse the implementation? moving it here?

Reply:
Sure

Reply:
Actually, looking at it a bit, I think it is better to defer that to when removing the `ActorMaterializer` in the future instead of introducing a bunch of logic around the also deprecated `ActorMaterializerSettings` here.